src/api_server.py

```diff
@@ -3,36 +3,47 @@
 from dataclasses import dataclass
 from datetime import datetime
 from pathlib import Path
-from typing import Dict, List
+from typing import Dict, List, Protocol
 
 import torch
 from fastapi import FastAPI
 from loguru import logger
 from pydantic import BaseModel, validator
 
-from src.model_extractors import BenefitsAndDrawbacksExtractor
+from src.model_extractors import Predictor, QuestionAnsweringPredictor, Seq2SeqPredictor
+
+PREDICTORS: Dict[str, Predictor] = {
+    "qa": classifier_qa,
+    "seq2seq": classifier_seq2seq,
+}
```

Comment:
This will not work as `classifier_qa` and `classifier_seq2seq` are not yet defined. Instead, set them to `None`:

```python
PREDICTORS: Dict[str, Predictor] = {
    "qa": None,
    "seq2seq": None
}
```